File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3dev/mathLogic-2.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os
import time

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
    },
    'sensors': {
    },
}
hal = Hal(_brickConfiguration)

___r1 = 0
___r2 = 0
___b1 = True
___r3 = 0
___sim = True
def run():
    global ___r1, ___r2, ___b1, ___r3, ___sim
    ___r3 = math.pi / float(2) if ( ___sim ) else 90
    ___b1 = ___b1 and ( math.sin(___r3) == 1 )
    ___b1 = ___b1 and ( math.cos(0) == 1 )
    ___b1 = ___b1 and ( math.tan(0) == 0 )
    ___b1 = ___b1 and ( math.asin(1) == ___r3 )
    ___b1 = ___b1 and ( math.acos(1) == 0 )
    ___b1 = ___b1 and ( math.atan(0) == 0 )
    ___b1 = ___b1 and ( ( math.e > float(2.6) ) and ( math.e < float(2.8) ) )
    ___b1 = ___b1 and ( ( ( math.sqrt(2) * math.sqrt(0.5) ) >= float(0.999) ) and ( ( math.sqrt(2) * math.sqrt(0.5) ) <= float(1.001) ) )
    # if b1 is true, the test succeeded, otherwise it failed

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/calliope2017NoBlue/mathLogic-2.cpp
#define _GNU_SOURCE

#include "MicroBit.h"
#include "NEPODefs.h"
#include <list>
#include <array>
#include <stdlib.h>
MicroBit _uBit;



double ___r1;
double ___r2;
bool ___b1;
double ___r3;
bool ___sim;

int main()
{
    _uBit.init();
    ___r1 = 0;
    ___r2 = 0;
    ___b1 = true;
    ___r3 = 0;
    ___sim = true;

    ___r3 = ( ( ___sim ) ? ( M_PI / ((float) 2) ) : ( 90) );
    ___b1 = ___b1 && ( sin(M_PI / 180.0 * (___r3)) == 1 );
    ___b1 = ___b1 && ( cos(M_PI / 180.0 * (0)) == 1 );
    ___b1 = ___b1 && ( tan(M_PI / 180.0 * (0)) == 0 );
    ___b1 = ___b1 && ( 180.0 / M_PI * asin(1) == ___r3 );
    ___b1 = ___b1 && ( 180.0 / M_PI * acos(1) == 0 );
    ___b1 = ___b1 && ( 180.0 / M_PI * atan(0) == 0 );
    ___b1 = ___b1 && ( ( M_E > 2.6 ) && ( M_E < 2.8 ) );
    ___b1 = ___b1 && ( ( ( M_SQRT2 * M_SQRT1_2 ) >= 0.999 ) && ( ( M_SQRT2 * M_SQRT1_2 ) <= 1.001 ) );
    // if b1 is true, the test succeeded, otherwise it failed
    release_fiber();
}

